tests/model_serving/maas_billing/multitenancy/aigateway/conftest.py

```
@@ -0,0 +1,188 @@
+
+
+@pytest.fixture
+def aigateway_adopting_preexisting_namespace(
+    admin_client: DynamicClient,
+    aigateway_infra_namespace: str,
+    teardown_resources: bool,
+) -> Generator[AIGatewayTestContext, Any, Any]:
+    """Create a tenant namespace first, then an AIGateway that adopts it with create=false."""
+    aigateway_name = f"e2e-aigw-adopt-ns-{generate_random_name()}"
+    tenant_namespace_name = tenant_namespace_name_for_aigateway(aigateway_name=aigateway_name)
+    with Namespace(client=admin_client, name=tenant_namespace_name, teardown=teardown_resources) as namespace:
```

Flake8 (7.3.0)

[error] local variable 'namespace' is assigned to but never used

(F841)